_query-dsl/regex-syntax.md

```diff
@@ -0,0 +1,312 @@
+- `$` – end of line
+
+Instead, your pattern must match the entire string to produce a match.
+
+## Example
+
+To try regular expressions, index the following documents into the `logs` index:
+
+```json
+PUT /logs/_doc/1
+{
+  "message": "error404"
+}
+```
+{% include copy-curl.html %}
+
+```json
+PUT /logs/_doc/2
+{
+  "message": "error500"
+}
+```
+{% include copy-curl.html %}
+
+```json
+PUT /logs/_doc/3
+{
+  "message": "error1a"
+}
+```
+{% include copy-curl.html %}
+
+### Example: Basic query containing regular expressions
+
+The following `regexp` query returns documents in which the entire value of the `message` field matches the pattern "error" followed by one or more digits. It does not match if the value only contains that pattern as a substring:
+
+```json
+GET /logs/_search
+{
+  "query": {
+    "regexp": {
+      "message": {
+        "value": "error[0-9]+"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+This query matches `error404` and `error500`:
+
+
+```json
+{
+  "took": 28,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 2,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": "logs",
+        "_id": "1",
+        "_score": 1,
+        "_source": {
+          "message": "error404"
+        }
+      },
+      {
+        "_index": "logs",
+        "_id": "2",
+        "_score": 1,
+        "_source": {
+          "message": "error500"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Example: Using optional operators
+
+The following query matches documents in which the `message` field exactly matches a string that starts with "error" followed by a number from 400 to 500, inclusive. The `INTERVAL` flag enables the use of `<min-max>` syntax for numeric ranges:
+
+```json
+GET /logs/_search
+{
+  "query": {
+    "regexp": {
+      "message": {
+        "value": "error<400-500>",
+        "flags": "INTERVAL"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+This query matches `error404` and `error500`:
+
+```json
+{
+  "took": 22,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 2,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": "logs",
+        "_id": "1",
+        "_score": 1,
+        "_source": {
+          "message": "error404"
+        }
+      },
+      {
+        "_index": "logs",
+        "_id": "2",
+        "_score": 1,
+        "_source": {
+          "message": "error500"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Example: Using ANYSTRING
+
+When the `ANYSTRING` flag is enabled, the `@` operator matches an entire string. This is useful when combined with intersection (`&`), because it allows you to construct queries that match full strings under specific conditions.
```

Comment:
```suggestion
When the `ANYSTRING` flag is enabled, the `@` operator matches an entire string. This is useful when combined with intersection (`&`) because it allows you to construct queries that match full strings under specific conditions.
```